gst/renderer/gstttmlrender.c

```diff
@@ -1093,6 +1093,7 @@ gst_ttml_render_text_chain (GstPad * pad, GstObject * parent,
       GST_TTML_RENDER_WAIT (render);
       GST_DEBUG ("Pad %s:%s resuming", GST_DEBUG_PAD_NAME (pad));
       if (render->text_flushing) {
+        gst_buffer_unref (buffer);
```

Comment:
@khwaaj, I found another leak ☝ Also picked your parser pad loop refactoring.
After it - no any leaks on my machine. Please, check it when you will have a chance.

Reply:
Sorry for the delay. Nice catch. I think the same issue may be present in the if-statements on lines 1049 and 1056, no?

I'm setting up to do some extended testing, but the issues around length expressions still exist. That implementation is entirely broken and doesn't even do what it is supposed to around the default length expression stuff.

Reply:
I have an alternate implementation of the length expression stuff where I simply use `nullptr` to represent the default expression instead, so it should not suffer from potential comparison problems, at least. I need to leave for a while but I'll try to synch our branches and apply this strategy once I'm back.

There are still things I do not like all over the place, e.g. that we keep the buffer reference in the scene even though we pass it on without upping the reference count (like I do on my branch), but that probably needs to be taken care of as a larger refactoring since it isn't really necessary in the first place.

Reply:
I rebased #16 on your branch, and made some additional changes:
1. Moved to using `nullptr` to represent the default length expression (which will hopefully help with the strange LE leaks you were seeing on Windows)
2. I added buffer unrefs to the two places I mentioned above

Please take it for a spin and see if it works for you.

Reply:
Checked your implementation - it works. 
Yeah, this trick with nullptr as the default length expression is really good and simplifies things👍 
Regarding keeping buffer reference inside the scene - agree, it should be refactored.

Reply:
Nice, I'll apply my patches to this PR and then maybe we can close this up.